projects/wave-core-new/src/lib/layers/layer.model.ts

```diff
@@ -30,72 +35,112 @@ export abstract class Layer implements ToDict<LayerDict> {
     }
 
     protected constructor(config: {
+        id?: number,
         name: string,
         workflowId: string,
+        isVisible: boolean,
+        isLegendVisible: boolean,
         symbology: AbstractSymbology,
     }) {
-        this._name = config.name;
+        this.id = config.id ?? Layer.nextLayerId++;
```

Comment:
is this for "replacing" an existing layer?

Reply:
Yes, you can either set an id or it is generated for you. This is necessary as long as the backend does not provide ids.